redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/YogaUIView.kt

```diff
@@ -94,24 +95,12 @@ internal class YogaUIView(
     }
   }
 
-  private fun calculateLayoutWithSize(size: CValue<CGSize>): CValue<CGSize> {
-    return size.useContents {
-      calculateLayoutWithSize(width.toYoga(), height.toYoga())
-    }
-  }
-
-  private fun calculateLayoutWithSize(width: Float, height: Float): CValue<CGSize> {
-    rootNode.requestedWidth = when (widthConstraint) {
-      Constraint.Fill -> width
-      else -> Size.UNDEFINED
-    }
-    rootNode.requestedMaxWidth = width
-
-    rootNode.requestedHeight = when (heightConstraint) {
-      Constraint.Fill -> height
-      else -> Size.UNDEFINED
-    }
-    rootNode.requestedMaxHeight = height
+  private fun calculateLayout(
+    width: Float = Size.UNDEFINED,
+    height: Float = Size.UNDEFINED,
+  ): CValue<CGSize> {
+    rootNode.requestedWidth = width
+    rootNode.requestedHeight = height
```

Comment:
Part of the fix. We now pass Size.UNDEFINED when we’re in measuring functions (sizeThatFits) but not when we’re in layout functions (`layoutSubviews()`).